pkg/logql/log/parser_test.go

```diff
@@ -87,6 +88,240 @@ func Test_jsonParser_Parse(t *testing.T) {
 	}
 }
 
+func TestJSONExpressionParser(t *testing.T) {
+	testLine := []byte(`{"app":"foo","field with space":"value","field with ÜFT8👌":"value","null_field":null,"bool_field":false,"namespace":"prod","pod":{"uuid":"foo","deployment":{"ref":"foobar", "params": [1,2,3]}}}`)
+
+	tests := []struct {
+		name        string
+		line        []byte
+		expressions []JSONExpression
+		lbs         labels.Labels
+		want        labels.Labels
+	}{
```

Comment:
Can we add a test against labels that already exist, such as:
```
"duplicate field"
[]byte(`{"foo": "bar"}`),
[]JSONExpression{NewJSONExpr("foo": "foo"},
labels.Labels{"foo": "bazz"}
labels.Labels{"foo": "bazz", "foo_extracted": "bar"}
```

Reply:
Where would `foo_extracted` come from?
`[]JSONExpression{NewJSONExpr("foo": "foo"},` would need to be `[]JSONExpression{NewJSONExpr("foo_extracted": "foo"},` in this case. Or am I missing something?

Does https://github.com/grafana/loki/pull/3280/files#diff-c3a5c0501669f6c25bb23430a1f1bb4fcce88bfccf3cfa6741122af4ad3e803aR254 go some way towards this?

Reply:
We have functionality that currently moves `<x>` to `<x>_extracted` if the label we extract from i.e. JSON is already present in the log stream's labels. It's how we deal with label collision during extraction.

See https://github.com/grafana/loki/blob/master/pkg/logql/log/parser.go#L123-L125

Reply:
Ah! I didn't actually know about that behaviour. Fixed